src/link_manager/LinkManagerStateMachineActiveActive.h

```diff
@@ -606,6 +615,35 @@ class ActiveActiveStateMachine : public LinkManagerStateMachineBase,
      */
     void setResetIcmpPacketCountsFnPtr(boost::function<void()> resetIcmpPacketCountsFnPtr) { mResetIcmpPacketCountsFnPtr = resetIcmpPacketCountsFnPtr; }
 
+    /**
+     * @method set
+     *
+     * @brief set mSendPeerProbeCommandFnPtr. This method is used for testing
+     *
+     * @param sendPeerProbeCommandFnPtr (in)           pointer to new sendPeerProbeCommandFnPtr
+     *
+     * @return none
+     */
+    void setSendPeerProbeCommandFnPtr(boost::function<void()> sendPeerProbeCommandFnPtr) { mSendPeerProbeCommandFnPtr = sendPeerProbeCommandFnPtr; }
+
+    /**
+     * @method startWaitMux
+     *
+     * @brief start waiting for mux, either mux set reply or probe reply
+     *
+     * @return none
+     */
+    inline void startWaitMux() { mWaitMux = true; }
+
+    /**
+     * @method stopWaitMux
+     *
+     * @brief stop waiting for mux, either mux set reply or probe reply
+     *
+     * @return none
+     */
+    inline void stopWaitMux() { mWaitMux = false; }
```

Comment:
Yeah I get that part, but we still do `setWaitStateCause`, then we should be able to `test` and `reset`, basically same as your `start` and `stop` here. 
https://github.com/sonic-net/sonic-linkmgrd/blob/5fd19a3dca0636a532517f870e5c2aa218b4bf30/src/link_manager/LinkManagerStateMachineActiveActive.cpp#L930

Reply:
The mux wait cause doesn't get cleaned up after receiving mux probe/notification or timing-out. So it is not safe to use it to check if we are waiting for mux probe/notification.

Reply:
There is a minor difference between `mWaitMux` and `WaitState::mWaitStateCause`:
* `mWaitMux` means that `linkmgrd` has requested a mux probe/set, and is in a timeout waiting state.
* `WaitState::mWaitStateCause` means that `linkmgrd` has requested a mux probe/set, and no response is received yet.

Let's say if `linkmgrd` has requested a mux set, and the mux wait timeouts and no response, `mWaitMux` will be cleared but `WaitState::mWaitStateCause` won't. So in this case, using `!mWaitMux` as a trigger to probe mux now will probe the mux, but using `WaitState::mWaitStateCause` won't.

Reply:
I think I see your point
`WaitState::mWaitStateCause` -> to tell if the response is unsolicited or not (if there is an unserved request) 
`mWaitMux` -> to avoid stressing the gRPC server or so (if there is a request we just send).